src/Repository/SpecialSponsorship/SpecialSponsorshipManager.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Repository\SpecialSponsorship;
+
+use App\Entity\SpecialSponsorship;
+use App\Dto\SpecialSponsorshipRequest;
+use Doctrine\ORM\EntityManagerInterface;
+use Ramsey\Uuid\Uuid;
+use Ramsey\Uuid\UuidInterface;
+
+final class SpecialSponsorshipManager implements SpecialSponsorshipManagerInterface
+{
+    private $entityManager;
+    private $repository;
+
+    public function __construct(EntityManagerInterface $entityManager)
```

Comment:
I'd probably just inject the SpecialSponsorship repository here (the interface used in it realistically), rather than the `$entityManager`. The EM is used for the save and remove methods, which I would definitely delegate to the repository itself, who already knows the EM anyway

Reply:
Should I remove "SpecialSponsorshipManager" class and use "SpecialSponsorshipRepository" with implements "SpecialSponsorshipRepositoryInterface" instead ?

Reply:
I would still have both, having the service consuming the repository, with an interface each, because I love encapsulation, but that's just me :)
Any opinion @lbassin?